torch/csrc/perf_tools/ThroughputBenchmark-inl.h

```diff
@@ -0,0 +1,143 @@
+#pragma once
+
+#include <random>
+#include <thread>
+
+#include <torch/csrc/utils/pybind.h>
+#include <torch/csrc/jit/pybind_utils.h>
+
+namespace torch {
+namespace throughput_benchmark {
+namespace detail {
+
+template <class Input, class Output, class Model>
+BenchmarkExecutionStats BenchmarkHelper<Input, Output, Model>::benchmark(
+    const BenchmarkConfig& config) const {
+  CHECK(initialized_);
+  TORCH_CHECK(
+      config.num_worker_threads == 1,
+      "Only parallelization by callers is supported");
+
+  // We pre-generate inputs for each of the threads and avoid threadsafety
+  // issues for random calls. As thread safe random is not provided in the
+  // standard library. We also replicate IValues. So we don't need to copy
+  // them during the main benchmark phase.
+  std::vector<std::vector<Input>> thread_inputs(config.num_calling_threads);
+  std::vector<size_t> input_iters(config.num_calling_threads);
+  {
+    std::random_device seeder;
+    std::mt19937 engine(seeder());
+    TORCH_CHECK(!inputs_.empty(), "Please provide benchmark inptus");
+    std::uniform_int_distribution<int> dist(0, inputs_.size() - 1);
+
+    for (int thread_id = 0; thread_id < config.num_calling_threads;
```

Comment:
looking at this, I'm curious if you want to set `FTZ` as an option or not in benchmark inputs. ill-conditioned inputs might definitely come in at runtime, and have really adverse perf impact without FTZ flag.

Reply:
@soumith, Hmmm. I don't think we ever did it in C2 production. 
One approach overal for providing knobs here I am thinking about - we will evolve things as we go and validate them on real use-cases. Otherwise I don't have a good use-case to validate effects of enabling FTZ as of right now. 

We also would need some sort of evaluation + benchmark setting in order to measure this. 

Thanks for bringing this up though! I added this to our benchmarking planning document.

I think in order to help to discover this opportunity for people, we can add it to the sweep tool. We can automatically find that FTZ is helpful and thus provide a recommendation. Then use can do their own evaluation with FTZ enabled.